tool/jt.rb

```diff
@@ -1777,21 +1778,26 @@ def build_native_image(*options)
     puts 'Building TruffleRuby native binary'
     chdir("#{graal}/substratevm") do
       File.write('mx.substratevm/env', "JAVA_HOME=#{java_home}\n")
-      mx 'build'
-
-      languages = %w[--Language:ruby]
-      if sulong
-        languages.unshift '--Language:llvm'
-        options = %w[
-          -Dtruffleruby.native.libsulong_dir=lib/cext/sulong-libs
-        ] + options
-      end
 
+      mx 'build'
       mx 'fetch-languages', '--Language:llvm', '--Language:ruby'
 
-      env = { "JAVA_HOME" => java_home }
-      output_options = "-H:Path=#{TRUFFLERUBY_DIR}/bin", '-H:Name=native-ruby', "-H:Class=org.truffleruby.Main"
-      raw_sh env, './native-image', '--no-server', *languages, *output_options, *options
+      languages = [*('--Language:llvm' unless no_sulong),
+                   '--Language:ruby']
+      options = [*('-Dtruffleruby.native.libsulong_dir=lib/cext/sulong-libs' unless no_sulong),
+                 *options]
```

Comment:
I don't like these expressions, they seem much more complex to me than the fairly straightforward code before.
`unless` in an array literal sounds quite of a hack to me.

Reply:
I prefer it this way, it describes what the array contains in one definition instead of having to track the state of the collections.

Reply:
But then it becomes hard to track what adding Sulong does and there is no easy case to read anymore (notably `options` is untouched if no sulong).
There are extra parens and the code adds `unless` at the end of lines which is easily missed (and you have to think `unless` returns `nil` if the condition is true, and that splatting nil returns `[]`: a lot of extra complexity), etc.
Also `options` is part of the expression so the full value is still not here.
I think `.unshift` is much clearer and expresses the intention to prepend an element to the Array (the original code should use `options.unshift` instead of of `options = %w[...] + options` though.

@nirvdrum Do you find the new code clearer?

Reply:
yeah for the options unshift is better. I prefer current definition of languages though.